packages/loader/container-definitions/src/runtime.ts

```diff
@@ -127,6 +127,8 @@ export interface IContainerContext extends IDisposable {
     readonly submitSignalFn: (contents: any) => void;
     readonly snapshotFn: (message: string) => Promise<void>;
     readonly closeFn: (error?: ICriticalContainerError) => void;
+    readonly createNextSummarizerFn:
```

Comment:
I think the callback is a good approach here, but wondering if you also consider event-based approaches (e.g. emit "becameSummarizer" or similar)?  Curious if you have thoughts there.